### ISSUE TITLE
Improper casting from parquet map to duckdb map via parquet_scan

### ISSUE THREAD

**Issue by ZaSkittles** [duckdb/duckdb#2752]
It's unclear to me if this has not been implemented or not, so apologies if this is a known issue:

I am writing a map column in my parquet files and trying to read them from duckdb using the python library.
When I query the files using pyarrow, the data looks correct and it sees it as a map:
```
dimensionMap: map<string, string> not null
  child 0, entries: struct<key: string not null, value: string not null> not null
      child 0, key: string not null
      child 1, value: string not null
```

However I cannot seem to actually extract data from the map easily:
When I am unable to use any of the built in map functions, as it seems the types aren't quite working as expected:
if trying to use the bracket notation `map['foo']` I get an error:
```
RuntimeError: Binder Error: No function matches the given name and argument types 'array_extract(LIST<STRUCT<key: VARCHAR, value: VARCHAR>>, VARCHAR)'. You might need to add explicit type casts.
```
and if trying to use `map_extract` directly I get an error :
```
RuntimeError: Binder Error: MAP_EXTRACT can only operate on MAPs
```

It seems some of the types are correct, but the queries don't seem to actually run as expected, the output of parquet_schema(...) for the relevant columns is given below:
![Screen Shot 2021-12-07 at 12 41 53 PM](https://user-images.githubusercontent.com/7540465/145103733-e15caaa9-0b05-48b9-9b17-c3afd1e770ad.png)

#### What happens?
Parquet map types of nested structs cannot be read by duckdb. 

#### To Reproduce
Steps to reproduce the behavior. Bonus points if those are only SQL queries.
I had to create an example file using pyarrow (I was unable to actually insert data using pyarrow since my project writes the data from the jvm).

```
import pyarrow as pa
import pyarrow.parquet as pq
import duckdb 
maptype = pa.map_(pa.string(), pa.string())
t = pa.Table.from_pydict({'my_map':[]}, pa.schema([pa.field('my_map',maptype)])) 
pq.write_table(t, 'test')
cursor = duckdb.connect()
cursor.execute("SELECT my_map['k'] FROM parquet_scan('test')").fetchall()
cursor.execute("SELECT map_extract(my_map,'k') FROM parquet_scan('test')").fetchall()
```

#### Environment (please complete the following information):
 - OS: iOS
 - DuckDB Version: 0.3.1
 - DuckDB Client: python

#### Before Submitting

- [X] **Have you tried this on the latest `master` branch?**
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.

- [X] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

**Comment by ZaSkittles** [comment]
This might be an issue specifically with the `parquet_scan` API. If I import the table directly from arrow it appears to work without issues. 
```
import pyarrow as pa
import pyarrow.parquet as pq
import duckdb 
t = pa.Table.from_pydict({'my_map':[]}, pa.schema([pa.field('my_map',maptype)])) 
arrow = duckdb.arrow(t)
arrow.query('arrow',"SELECT my_map['k'] FROM arrow").fetchall()
arrow.query('arrow',"SELECT map_extract(my_map,'k') FROM arrow").fetchall()
```

**Comment by Mytherin** [comment]
Indeed, maps were being read as structs of lists (which is how maps are stored internally) rather than as proper maps. I have pushed a fix in #2838 with extended map support for the Parquet reader/writer, which should fix this issue.
